### PR TITLE
Don't run collections tests against CBS versions that don't support them

### PR DESCRIPTION
**Pull request by markspolakovs**
## Dependencies

- [x] #5641 

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `Server 6.6.5, package rest` https://jenkins.sgwdev.com/job/SyncGateway-Integration/456/
  * [TestReplicationConcurrentPush flake](https://issues.couchbase.com/browse/CBG-1971)

### CODE REVIEW

File: base/util_testing.go [hunk 1]
```diff
@@ -670,3 +670,33 @@ func ForAllDataStores(t *testing.T, testCallback func(*testing.T, sgbucket.DataS
 		})
 	}
 }
+
+// clusterCompatibility = MajorVersion * 65536 + MinorVersion
+const minCompatVersionForCollections = 7 * 65536
```
Comment (bbrks) [inline base/util_testing.go]: Might be worth seeing if we can use `decodeClusterVersion()`/`encodeClusterVersion()` instead (in `base/util.go`)

File: base/main_test_cluster.go [hunk 2]
```diff
@@ -271,3 +272,20 @@ func (c *tbpClusterV2) closeCluster(cluster *gocb.Cluster) {
...
+	defer c.closeCluster(cluster)
+
+	nodesMeta, err := cluster.Internal().GetNodesMetadata(nil)
+	if err != nil {
+		FatalfCtx(context.Background(), "TEST: failed to fetch nodes metadata: %v", err)
+	}
+	minVal := math.MaxInt
+	for _, node := range nodesMeta {
+		if node.ClusterCompatibility < minVal {
+			minVal = node.ClusterCompatibility
```
Comment (bbrks) [inline base/main_test_cluster.go]: Isn't `ClusterCompatibility` already `min(version)` across all nodes in the cluster? Do we have to check each node?

In other areas we've always just got the first node's cluster compat version for this reason

File: base/util_testing.go [hunk 3]
```diff
@@ -670,3 +670,33 @@ func ForAllDataStores(t *testing.T, testCallback func(*testing.T, sgbucket.DataS
...
+
+// map of server names to cluster compat versions (string->int)
+var serverClusterCompat sync.Map
```
Comment (bbrks) [inline base/util_testing.go]: I'm not sure I understand the intention of the `c.server` lookup? There's only one server behind a test bucket pool

**Review by bbrks — CHANGES_REQUESTED** [review]

File: base/main_test_cluster.go (hunk 2 above)
Comment (markspolakovs) [reply base/main_test_cluster.go]: Ah, didn't know that!

**Review by bbrks — APPROVED** [review]
LGTM pending rebase/conflict fix

**Pull request merged**